src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,70 @@
-import React from 'react';
+import { useEffect, useState } from 'react';
+import cn from 'classnames';
 import { Loader } from '../Loader';
+import { Todo, User } from '../../types';
+import { useTodos } from '../../app/hooks';
+import { getUser } from '../../api';
 
 export const TodoModal: React.FC = () => {
+  const { currentTodo, handleResetCurrentTodo } = useTodos();
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const { id, title, completed, userId } = currentTodo as Todo;
```

Comment:
The use of `currentTodo as Todo` for type assertion can lead to runtime errors if `currentTodo` is `null` or `undefined`. Consider adding a check to ensure `currentTodo` is not `null` before destructuring, or use optional chaining to safely access properties.